src/c-writer.cc

```diff
@@ -67,18 +68,36 @@ struct Label {
   bool used = false;
 };
 
-template <int>
-struct Name {
-  explicit Name(const std::string& name) : name(name) {}
+struct LocalName {
+  explicit LocalName(const std::string& name) : name(name) {}
   const std::string& name;
 };
 
-using LocalName = Name<0>;
-using GlobalName = Name<1>;
-using ExternalPtr = Name<2>;
-using ExternalRef = Name<3>;
-using ExternalInstancePtr = Name<4>;
-using ExternalInstanceRef = Name<5>;
+template <ModuleFieldType>
```

Comment:
What is the purpose of making this a template?   It doesn't seem to use the template parameter at all.

Reply:
I wanted all the GlobalNames to be different types (templated on the index space / module field) so that `Write(const GlobalName<T>&)` could also be templated (based on the type of GlobalName) so it can look up the correctly suffixed name in the `global_sym_map_`.

If you prefer, we could just have every GlobalName include a member variable of type `ModuleFieldType`, and then `Write(const GlobalName&)` examines that member. Then there's no templates, at the cost of a tiny amount of runtime overhead.

Reply:
Ah I think I understand now.. I personally think I would find that form more readable.  I guess I'm slightly biased against template magic though.   What do you think?   I won't block this PR as it stands if you feel its worth the trade-off.

Reply:
I really do not feel strongly. :-) Here it is the way you suggested.